src/llama-chat.cpp

```diff
@@ -183,6 +183,11 @@ llm_chat_template llm_chat_detect_template(const std::string & tmpl) {
         return LLM_CHAT_TEMPLATE_BAILING;
     } else if (tmpl_contains("<|header_start|>") && tmpl_contains("<|header_end|>")) {
         return LLM_CHAT_TEMPLATE_LLAMA4;
+    } else if (tmpl_contains("<|userprompt|>") &&
+               tmpl_contains("<|endofuserprompt|>") &&
+               tmpl_contains("<|response|>") &&
+               tmpl_contains("<|endofresponse|>")) {
+        return LLM_CHAT_TEMPLATE_DOTS1;
```

Comment:
```suggestion
    } else if (tmpl_contains("<|endofuserprompt|>")) {
        return LLM_CHAT_TEMPLATE_DOTS1;
```

we don't need to check exhaustively since this is the only model which uses this  `<|endofuserprompt|>` marker